### PR TITLE
Create a stable S3 path to the current issue

### PR DESCRIPTION
**Pull request by hosamaly**
We configure Amazon's KPP with a stable URL that should point to each day's data.
This commit enables us to have a stable path within the S3 bucket via a redirect.

The redirect is recreated every time the Lambda is run to point to the generated
files. For example, the following link:
http://my-bucket.s3-website-eu-west-1.amazonaws.com/CODE/CurrentIssue/hierarchical-title-manifest.xml
will point to:
http://my-bucket.s3-website-eu-west-1.amazonaws.com/CODE/2018-07-02/hierarchical-title-manifest.xml

### CODE REVIEW

File: src/main/scala/com/gu/kindlegen/app/Lambda.scala [hunk 1]
```diff
@@ -113,22 +117,41 @@ class Lambda(settings: Settings, date: LocalDate) extends Logging {
   def doRun(remainingTimeInMillis: => Long): Unit = {
     import scala.concurrent.ExecutionContext.Implicits.global
 
+    val publisher = s3Publisher(settings)
+    logger.info(s"Starting to publish files for $date; uploading to s3://${settings.s3.absolutePath.source}")
+
+    val deletePublicDir = publisher
```
Comment (regiskuckaertz) [inline src/main/scala/com/gu/kindlegen/app/Lambda.scala]: Can you run actions in parallel?

File: src/main/scala/com/gu/kindlegen/app/Lambda.scala [hunk 2]
```diff
@@ -113,22 +117,41 @@ class Lambda(settings: Settings, date: LocalDate) extends Logging {
...
+      .undirect(settings.s3.publicDirectory)
+      .andThen { case Failure(error) => logger.warn("Deleting the public directory failed!", error) }
+
+    val generateFiles = deletePublicDir
```
Comment (regiskuckaertz) [inline src/main/scala/com/gu/kindlegen/app/Lambda.scala]: I don't think `generateFiles` should wait on `deletePublicDir`, but `publishPublicly` should. wdyt?

File: src/main/scala/com/gu/kindlegen/app/Lambda.scala (hunk 1 above)
Comment (hosamaly) [reply src/main/scala/com/gu/kindlegen/app/Lambda.scala]: We want to make sure that Amazon won't be seeing partial content, so we want the operation to fail if this step fails. This is what we agreed upon, right @tomrf1?

Comment (regiskuckaertz) [reply src/main/scala/com/gu/kindlegen/app/Lambda.scala]: Oh I see 👍

**Pull request merged**